e_os.h

```diff
@@ -303,6 +303,54 @@ struct servent *getservbyname(const char *name, const char *proto);
 # endif
 /* end vxworks */
 
+/* system-specific variants defining ossl_sleep() */
+#ifdef OPENSSL_SYS_UNIX
+# include <unistd.h>
+static ossl_inline void ossl_sleep(unsigned long millis)
+{
+# ifdef OPENSSL_SYS_VXWORKS
+    struct timespec ts;
+    ts.tv_sec = (long int) (millis / 1000);
+    ts.tv_nsec = (long int) (millis % 1000) * 1000000ul;
+    nanosleep(&ts, NULL);
+# elif defined(__TANDEM)
+#  if !defined(_REENTRANT)
+#   include <cextdecs.h(PROCESS_DELAY_)>
+    /* HPNS does not support usleep for non threaded apps */
+    PROCESS_DELAY_(millis * 1000);
+#  elif defined(_SPT_MODEL_)
+#   include <spthread.h>
+#   include <spt_extensions.h>
+    usleep(millis * 1000);
+#  else
+    usleep(millis * 1000);
+#  endif
+# else
+    usleep(millis * 1000);
+# endif
```

Comment:
Wouldn't it be easier to have a single branch?
```suggestion
# elif defined(__TANDEM) && !defined(_REENTRANT)
#  include <cextdecs.h(PROCESS_DELAY_)>
   /* HPNS does not support usleep for non threaded apps */
   PROCESS_DELAY_(millis * 1000);
# elif defined(__TANDEM) && defined(_SPT_MODEL_)
#  include <spthread.h>
#  include <spt_extensions.h>
   usleep(millis * 1000);
# else
    usleep(millis * 1000);
# endif
```

Reply:
Yes, this would be equivalent and a little more compact.